src/coreclr/interpreter/compiler.cpp

```diff
@@ -2130,6 +2130,42 @@ int32_t InterpCompiler::GetDataItemIndexForHelperFtn(CorInfoHelpFunc ftn)
     return GetDataItemIndex(addr);
 }
 
+bool InterpCompiler::EmitNamedIntrinsicCall(NamedIntrinsic ni, CORINFO_CLASS_HANDLE clsHnd, CORINFO_METHOD_HANDLE method, CORINFO_SIG_INFO sig)
+{
+    bool mustExpand = (method == m_methodHnd);
+    if (!mustExpand && (ni == NI_Illegal))
+        return false;
+
+    switch (ni)
+    {
+        case NI_IsSupported_False:
+        case NI_IsSupported_True:
+            AddIns(INTOP_LDC_I4);
+            m_pLastNewIns->data[0] = ni == NI_IsSupported_True;
+            PushStackType(StackTypeI4, NULL);
+            m_pLastNewIns->SetDVar(m_pStackPointer[-1].var);
+            return true;
+
+        case NI_Throw_PlatformNotSupportedException:
+            // Interpreter-FIXME: If an INTOP_THROW_PNSE is the first opcode in a try block, catch doesn't catch the exception
+            // For now, insert a safepoint as padding.
+            AddIns(INTOP_SAFEPOINT);
+            AddIns(INTOP_THROW_PNSE);
+            return true;
```

Comment:
Should this include implementation of more of the intrinsics that we recognize? Or at least establish the pattern.

Is the idea that each recognized intrinsic will get its own opcode?

Reply:
I can do that. Do you see it as a barrier to landing this initial PR? I wanted to avoid it getting too big since it will bitrot easily like i.e. the pinvoke PR.